modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java

```diff
@@ -683,63 +676,17 @@ private FieldsResult executeFieldsQuery(CacheQuery<?> qry, @Nullable Object[] ar
                     args,
                     securitySubjectId(cctx),
                     taskName));
-            }
-
-            // Attempt to get result from cache.
-            resKey = new T2<>(qry.clause(), F.asList(args));
-
-            res = (FieldsResult)qryResCache.get(resKey);
-
-            if (res != null && res.addRecipient(rcpt))
-                return res; // Cached result found.
-
-            res = new FieldsResult(rcpt);
-
-            if (qryResCache.putIfAbsent(resKey, res) != null)
-                resKey = null; // Failed to cache result.
         }
-        else {
-            assert qry.type() == SPI : "Unexpected query type: " + qry.type();
-
-            if (cctx.events().isRecordable(EVT_CACHE_QUERY_EXECUTED)) {
-                cctx.gridEvents().record(new CacheQueryExecutedEvent<>(
-                    cctx.localNode(),
-                    "SPI query executed.",
-                    EVT_CACHE_QUERY_EXECUTED,
-                    CacheQueryType.SPI.name(),
-                    cctx.name(),
-                    null,
-                    null,
-                    null,
-                    null,
-                    args,
-                    securitySubjectId(cctx),
-                    taskName));
-            }
 
-            res = new FieldsResult(rcpt);
-        }
+        // Attempt to get result from cache.
+        T2<String, List<Object>> resKey = new T2<>(qry.clause(), F.asList(args));
 
-        try {
-            if (qry.type() == SPI) {
-                IgniteSpiCloseableIterator<?> iter = cctx.kernalContext().indexing().query(cacheName, F.asList(args),
-                    filter(qry));
+        FieldsResult res = (FieldsResult) qryResCache.get(resKey);
```

Comment:
remove whitespace after (FieldsResult). This and some other issues will fail if you run build with enabled "checkstyle" profile.